### ISSUE TITLE
Massive Termination of  Runaway Scripts

### ISSUE THREAD

**Issue by Tranthula** [Kyle873/DoomRPG#95]
I've been playing this for about a month, and these scripts are putting my progress to a halt. I've tried my best to reproduce this bug but I fail. I'm meshing DRPG and DRLA.
Here's the chunk of scripts straight from the console (I'll give a description of what all happens when I get this bug below)

Runaway script "MonsterInit" Terminated (x8)
Runaway script "GetAuraTokens" Terminated
Runaway script "Monsterinit" Terminated (x8)
Runaway script "GetAuraTokens" Terminated
Runaway script "Monsterinit" Terminated (x38)
( ^ This repeats for a bit down the console)
Runaway script "_GlobalInit" Terminated
Runaway script "MapInit" Terminated
Runaway script "_PassingEvents" Terminated
Runaway script "_ShopSpecialHandler" Terminated
Runaway script "_StimHUD" Terminated
Runaway script "_EventHUD" Terminated
Runaway script "_PowerupHUD" Terminated
Runaway script "_DRLAHUD" Terminated
Runaway script "_ComboHUD" Terminated
Runaway script "_CoopViewHUD" Terminated
Runaway script "_Init" Terminated
Runaway script "_AuraTimeHUD" Terminated
Runaway script "_StatusEffectHUD" Terminated
Runaway script "_MultiplayerHUD" Terminated
Runaway script "_OverviewHUD" Terminated
Runaway script "_StatHUD" Terminated
Runaway script "_TurrentHUD" Terminated
Runaway script "_MissionHUD" Terminated
Runaway script "_TurretCommandWheel" Terminated
Runaway script "_SkillHUD" Terminated
Runaway script "_SkillWheel" Terminated
Runaway script "_TurretLoopMaintenance" Terminated
Runaway script "_TurretLoop" Terminated

Wow, sorry for the wall of text. I was looking at the other issues reported and none of them had scripts like this.
So, basically, when this happens, every HUD except for GZDoom is missing. Doing Hud Preview doesn't work, and I can't access any menus or the shop. When I get to the Outpost, none of the marines will retaliate when attacked. When I Save Character, I'm given the message that I have to be in the outpost to save, even though I'm already there. When I access the teleporter, I can scroll through the maps, but accessing the one I came from doesn't let me and tells me I'm already there.

So, apologies if I'm jumping on some sort of 'fix this' bandwagon, that's not my intention. I love this mod(s), the work you all put in is appreciated. Thanks!

**Comment by subenji** [comment]
I've been trying to debug this issue for the past week!
This is currently the focus of my attention.

**Comment by Tranthula** [comment]
Assumed someone would be looking at this. If you need me to do anything, I'm more than willing to help.

**Comment by Tranthula** [comment]
My brother has a savefile of this bug. At level change, termination happens, no matter what level he goes to.

**Comment by subenji** [comment]
Unfortunately, at that point it's too late - I need to catch it in the act as it were. I've been trying to reproduce it reliably with no luck, and my attempts to catch it with a debugger attached to gzdoom have been fruitless - not been able to reproduce it when debugging!
A new version of the GDCC compiler was just released, and I've just updated my fork with a version built with it. It's possible this could ahve fixed it, either way it's going to need retesting. As I'm struggling to reproduce it I'd ask if you can do the same, and when it happens if you could give me the earliest autosave you have, that should give me everything I need.

**Comment by subenji** [comment]
A forum member noticed this only happens when playing Kill Reinforcements missions, and so I was able to trace and fix this.

**Comment by Tranthula** [comment]
I've noticed this also! Props on grabbing a fix; but sadly, I've encountered the error once more. And yes, I was playing a Kill Reinforcements mission but with the Armageddon anomaly. It only happened once, so I'll keep an eye out for this bug while taking on anomalies.
